clouddrift/transfer.py

```diff
@@ -0,0 +1,843 @@
+        else:
+            # Mixed solution
+            coeff = 1 / (density * sqrt(1j * (omega + coriolis_frequency) * K0))
+            G = coeff * kv(0, xiz) / kv(1, xi0)
+    else:
+        if K1 == 0:
+            # Finite-layer Ekman
+            coeff = 1 / (density * sqrt(1j * (omega + coriolis_frequency) * K0))
+            numer = np.sinh((1 + 1j) * (bld - z) / delta1)
+            denom = np.cosh((1 + 1j) * bld / delta1)
+            G = coeff * numer / denom
+        elif K0 == 0:
+            # Finite-layer Madsen
+            coeff = 2 / (density * K1)
+            k0z = kv(0, 2 * sqrt(1j * z / delta2))
+            k0h = kv(0, 2 * sqrt(1j * bld / delta2))
+            i0z = iv(0, 2 * sqrt(1j * z / delta2))
+            i0h = iv(0, 2 * sqrt(1j * bld / delta2))
+            G = coeff * (k0z - (k0h * i0z / i0h))
+        else:
+            # Finite-layer mixed solution
+            coeff = 1 / (density * sqrt(1j * (omega + coriolis_frequency) * K0))
+            k0z, i0z, k0h, i0h, k10, i10 = bessels_noslip(xiz, xih, xi0)
+            numer = i0h * k0z - k0h * i0z
+            denom = i10 * k0h + k10 * i0h
+            G = coeff * numer / denom
+
+    return G
+
+
+def _wind_transfer_elipot_free_slip(
+    omega: np.ndarray,
+    z: np.ndarray,
+    coriolis_frequency: float,
+    delta: float,
+    mu: float,
+    bld: float,
+    density: float,
+) -> np.ndarray:
+    """
+    Transfer function from wind stress to oceanic velocity with free-slip boundary condition, Elipot method.
+    """
+
+    zo = np.divide(delta**2, mu)
+    K0 = 0.5 * delta**2 * np.abs(coriolis_frequency)
+    K1 = 0.5 * mu * np.abs(coriolis_frequency)
+    delta1 = sqrt(2 * K0 / (omega + coriolis_frequency))
+    delta2 = K1 / (omega + coriolis_frequency)
+    xiz = 2 * sqrt(1j * (zo + z) / delta2)
+    xi0 = 2 * sqrt(1j * zo / delta2)
+    xih = 2 * sqrt(1j * (zo + bld) / delta2)
+
+    if K1 == 0:
+        # Finite-layer Ekman
+        coeff = _rot(-np.pi / 4) / (
+            density * sqrt(1j * (omega + coriolis_frequency) * K0)
+        )
+        numer = np.cosh((1 + 1j) * (bld - z) / delta1)
+        denom = np.sinh((1 + 1j) * bld / delta1)
+        G = coeff * numer / denom
+    elif K0 == 0:
+        # Finite-layer Madsen
+        coeff = 2 / (density * K1)
+        k0z = kv(0, 2 * sqrt(1j * z / delta2))
+        k0h = kv(0, 2 * sqrt(1j * bld / delta2))
+        i0z = iv(0, 2 * sqrt(1j * z / delta2))
+        i1h = iv(1, 2 * sqrt(1j * bld / delta2))
+        G = coeff * (k0z - (k0h * i0z / i1h))
+    else:
+        # Finite-layer mixed solution
+        coeff = 1 / (density * sqrt(1j * (omega + coriolis_frequency) * K0))
+        k1h = kv(1, xih)
+        i0z = iv(0, xiz)
+        i1h = iv(1, xih)
+        k10 = kv(1, xi0)
+        i10 = iv(1, xi0)
+        k0z = kv(0, xiz)
+        numer = k1h * i0z + i1h * k0z
+        denom = i1h * k10 - i10 * k1h
+        G = coeff * numer / denom
+
+    return G
+
+
+def bessels_freeslip(
```

Comment:
Do we want those part of the API of the modules? Or they should start with `_`.

Reply:
they should indeed.